src/main/java/Duke.java

```diff
@@ -1,10 +1,128 @@
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printWelcomeMessage();
+
+        Scanner in = new Scanner(System.in);
+        String input = in.nextLine();
+
+        while (!input.equalsIgnoreCase(EXIT_COMMAND)) {
+            System.out.println(DOTTED_LINE);
+            if (input.equalsIgnoreCase(LIST_COMMAND)) {
+                handleInput(input, LIST_COMMAND);
+            } else if (input.startsWith(DONE_COMMAND)) {
+                handleInput(input, DONE_COMMAND);
+            } else if (input.startsWith(TODO_COMMAND)) {
+                handleInput(input, TODO_COMMAND);
+            } else if (input.startsWith(DEADLINE_COMMAND)) {
+                handleInput(input, DEADLINE_COMMAND);
+            } else if (input.startsWith(EVENT_COMMAND)) {
+                handleInput(input, EVENT_COMMAND);
+            } else {
+                handleInput(input, "");
+            }
+            System.out.println(DOTTED_LINE);
+            input = in.nextLine();
+        }
+
+        printEndMessage();
+    }
+
+    private static void printWelcomeMessage() {
+        System.out.println(LOGO);
+        System.out.println(DOTTED_LINE);
+        System.out.println("\tHello! I'm Duke");
+        System.out.println("\tWhat can I do for you?");
+        System.out.println(DOTTED_LINE);
+    }
+
+    private static void handleInput(String input, String command) {
+        switch (command) {
+        case LIST_COMMAND: {
+            printList();
+            break;
+        }
+        case DONE_COMMAND: {
+            int index = Integer.parseInt(input.substring(command.length()).trim());
+            Task task = tasks[index - 1];
+            markTaskAsDone(task);
+            break;
+        }
+        case TODO_COMMAND: {
+            String name = input.substring(TODO_COMMAND.length()).trim();
+            Task todo = new Todo(name);
+            addToTaskList(todo);
+            break;
+        }
+        case DEADLINE_COMMAND: {
+            int byIndex = input.indexOf(BY_COMMAND);
+            String name = input.substring(DEADLINE_COMMAND.length(), byIndex).trim();
+            String by = input.substring(byIndex + BY_COMMAND.length()).trim();
+            Task deadline = new Deadline(name, by);
+            addToTaskList(deadline);
+            break;
+        }
+        case EVENT_COMMAND: {
+            int atIndex = input.indexOf(AT_COMMAND);
+            String name = input.substring(EVENT_COMMAND.length(), atIndex).trim();
+            String at = input.substring(atIndex + AT_COMMAND.length()).trim();
+            Task event = new Event(name, at);
+            addToTaskList(event);
+            break;
+        }
+        default: {
+            Task task = new Task(input);
+            addToTaskList(task);
+            break;
+        }
```

Comment:
Perhaps the opening and closing braces of the default clause should be removed also to follow coding standards.